uw-frame-components/portal/main/controllers.js

```diff
@@ -81,39 +81,60 @@ define(['angular', 'require'], function(angular, require) {
 
   /* Username */
   .controller('SessionCheckController',
-  ['$log', '$scope', 'mainService', 'NAMES',
+  ['$log', '$scope', 'mainService', 'portalShibbolethService', 'NAMES',
   'FOOTER_URLS', '$sessionStorage', '$rootScope',
-  function($log, $scope, mainService, NAMES,
+  function($log, $scope, mainService, portalShibbolethService, NAMES,
   FOOTER_URLS, $sessionStorage, $rootScope) {
     var vm = this;
     vm.user = [];
+    vm.username = '';
+    vm.campusId = '';
     vm.firstLetter = '';
 
     $scope.FOOTER_URLS = FOOTER_URLS;
     $scope.usernameOptionOpen = false;
 
+    // Get the user's attributes from Shibboleth
+    portalShibbolethService.getUserAttributes()
+      .then(function(result) {
+        angular.forEach(result, function(key) {
+          // Set username
+          if (key.name === 'displayName') {
+            var username = key.values[0];
+            if (username === '' || !angular.isString(username)) {
+              vm.username = '?';
+              vm.firstLetter = '?';
+            } else {
+              vm.username = username;
+              vm.firstLetter = username.substring(0, 1);
+            }
+          }
+          // Set campus ID
```

Comment:
Need to gate this behind a configuration flag so that via configuration we can turn this on or off. :/